components/kubeflow/deployer/src/deploy.sh

```diff
@@ -95,6 +105,17 @@ echo "Generating the TF Serving config..."
 ks generate tf-serving server --name="${SERVER_NAME}"
 ks param set server modelPath "${MODEL_PATH}/export/export"
 
+# support local nfs pvc to deploy tf-serving.
+if [ "x${MODE_STORAGE_TYPE}" == "xnfs" ];then
+  if [ ! -z "${PVC_NAME}" ];then
+    ks param set server modelStorageType "${MODE_STORAGE_TYPE}"
```

Comment:
Why do we need `MODE_STORAGE_TYPE`? Also Kubernetes support many different volume types https://kubernetes.io/docs/concepts/storage/#types-of-volumes , do you want to handle all of the cases here?

I think here what we need is only a PVC, as long as we have the PVC, the pod can use this PVC directly and do not need to care where the PV from.

Reply:
Thanks for your comments, the reason for the specified MODE_STORAGE_TYPE is to discriminate local storage from cloud storage. The pvc defination makes sence only if the storage type is local mode. I think that's good way to have one parameter to define this to avoid any mistake for cloud storage which is used in the sample. 
You can see more from the kubeflow document: [Serve a local model using Tensorflow Serving](https://github.com/kubeflow/kubeflow/blob/master/components/k8s-model-server/Serve%20a%20local%20model%20using%20Tensorflow%20Serving.md).

Reply:
Apologies if speaking out of turn. We have looked extensively into PVC support when refactoring the notebook spawner UI, so I'd like to understand this better.

I agree with @gyliu513:  Whenever we need to interface with K8s storage resources, we just use a PVC. Why should the code know if the PVC came from NFS or from any other storage type, will it do something differently?

If the code really needs to know whether it is going to use the PVC parameter or not, then OK, maybe there could be one more parameter, but then its value should be, e.g., `local` vs. `cloud`, not `nfs`. But this is not consistent, I should be able to use a PVC on the cloud, as well.

So, I think the simplest way would be: If a PVC has been defined, use it. If not, don't. Will this not make the `MODE_STORAGE_TYPE` argument unnecessary then, and simplify the implementation?